include/cantera/base/logger.h

```diff
@@ -67,6 +67,20 @@ class Logger
         std::cout << std::endl;
     }
 
+    //! Write a warning message.
+    /*!
+     * The default behavior is to write to the logging output.
+     * @param msg      String message to be written to cout
+     */
+    virtual void warn(const std::string& msg, const std::string& warning) {
```

Comment:
I noticed that you've got the message as the first argument and the type of warning second. This is the opposite of the `Py_*` functions, so it'd be nice to be consistent. Unless, of course, this is consistent with other Cantera functions, in which case, it should stay 😄

Reply:
Good point. I switched the order around and created a more generic warning type (starting in `global.h`).